tests/integration/MultiSigStateMachine/TUTORIAL.md

```diff
@@ -141,90 +141,32 @@ write the equivalent command line for your contract.
-builtin to represent pattern matching.  A destructor for a type is the
-opposite of a constructor. It characterizes the use of the type's
-inhabitants.  As a concrete example, the `maybe` function in Haskell
-is the destructor (or eliminator) for the `Maybe` datatype.  In
-PlutusIR there is no pattern-matching hence we must rely on destructors
-to do case analisys.
-
-The `--pull-destr n` (or `-p n`) option will instruct pirouette to pull the `n`-th destructor
-to the top of the program, if possible. Because we are receiving generated code, we never know for sure
-which value the plutus decided to pattern-match on first. We can, however, interactively discover this.
-We can ask pirouette to print its _terms_ and inspect what we're dealing with by running:
+The `--with-args param,st,INPUT` option tells pirouette to pass the given list of free names
+to the transition function and tells pirouette which of those parameters represents the user input,
+through the `INPUT` identifier. Under the hood, pirouette eta-expands the underlying program with:
 
 ```
-cabal exec pirouette -- MultiSigStateMachine.flat --prefix transition --term-only | less
+transition' param st INPUT
+  = case INPUT of
+      Cons1 a1 ... aN -> transition param st (Cons1 a1 ... aN)
+      ...
+      ConsK a1 ... aM -> transition param st (ConsK a1 ... aM)
 ```
 
-Here, we are instructing pirouette to load the definitions from `MultiSigStateMachine.flat`, filter all of those whose name contains `transition` as a prefix and print their definitions. In this case, we should see a lot of output, the beginning of which is:
+Then we normalize the resulting program and produce TLA+ code out of it.
+This ensures that the first thing that happens is the pattern matching on the variable representing
+the contract's endpoints.
 
-```
-transition :=
-  Params0 -> (State0 MSState) -> Input -> (Maybe (Tuple20 (TxConstraints0 Void
-                                                                          Void)
-                                                          (State0 MSState)))
-  λ (params1098 : Params0) (ds1099 : State0 MSState) (i1100 : Input)
-  . State_match @MSState
-      1#ds1099
-      @(Maybe (Tuple20 (TxConstraints0 Void Void) (State0 MSState)))
-      (λ (ds1101 : MSState) (ds1102 : List0 (Tuple20 ByteString
-                                                     (List0 (Tuple20 ByteString
-                                                                     Integer))))
-       . MSState_match 1#ds1101
-           @(Maybe (Tuple20 (TxConstraints0 Void Void) (State0 MSState)))
-           (λ (pmt1107 : Payment0) (pks1108 : List0 ByteString)
-            . Input_match 4#i1100
-                @(Maybe (Tuple20 (TxConstraints0 Void Void) (State0 MSState)))
-  ...
-```
-
-This is a simplification of the PlutusIR representation of the
-`transition` function and is what pirouette is going to use
-when extracting the TLA+ model from it.  Yet, the first
```

Comment:
I think that from line 157 to this point should be kept, either at the end of this file or in a file called `Internal.md` referenced in the read me.

Reply:
Makes sense! I'll keep it and move it to the end of the file